source/_UIAHandler.py

```diff
@@ -224,9 +230,40 @@ def MTAThreadFunc(self):
 			self.MTAThreadInitException=e
 		finally:
 			self.MTAThreadInitEvent.set()
-		self.MTAThreadStopEvent.wait()
+		while True:
+			func=self.MTAThreadQueue.get()
+			if func:
+				func()
+			else:
+				break
 		self.clientObject.RemoveAllEventHandlers()
 
+	def _onForegroundChange(self):
+		pendingForegroundUIAElement=self.pendingForegroundUIAElement
+		if pendingForegroundUIAElement==self.currentForegroundUIAElement:
+			return
+		if self.currentForegroundUIAElement:
+			try:
+				self.clientObject.removePropertyChangedEventHandler(self.currentForegroundUIAElement,self)
+			except COMError:
+				# The old UIAElement died as the window was closed.
+				# The system should forget the old event registration itself.
+				pass
+		self.currentForegroundUIAElement=pendingForegroundUIAElement
+		try:
+			self.clientObject.AddPropertyChangedEventHandler(self.currentForegroundUIAElement,TreeScope_Subtree,self.baseCacheRequest,self,UIAPropertyIdsToNVDAEventNames.keys())
+		except COMError:
+			log.error("Could not register for UIA property change events for new foreground")

```

Comment:
Re me previous comment, I assume this is only problematic for native UIA windows?

Reply:
problematic for any descendant native UIA windows, yes. Again, the foreground window itself may not natively support UIA. However, the bridged UIA element we get for the foreground window in this case is still enough for UIA core to register propertyChange events for the subtree.